tests/src/main/java/com/hazelcast/stabilizer/tests/helpers/HazelcastTestUtils.java

```diff
@@ -167,16 +167,32 @@ public static HazelcastInstanceImpl getHazelcastInstanceImpl(HazelcastInstance h
         return impl;
     }
 
-    public static long nextKeyOwnedBy(long key, HazelcastInstance instance) {
-        final Member localMember = instance.getCluster().getLocalMember();
-        final PartitionService partitionService = instance.getPartitionService();
-        for (; ; ) {
+    /**
+     * Returns the next {@code long} key owned by the given Hazelcast instance.
+     * <p/>
+     * If no key is found after 100 retries a {@link RuntimeException} is thrown.
+     *
+     * @param instance  Hazelcast instance to search next key for
+     * @param lastKey   last key to start search from
+     * @param increment increment to add if next key did not belong to given instance
+     * @return next key owned by given Hazelcast instance
+     * @throws RuntimeException if no key could be found after 100 retries
+     */
+    public static long nextKeyOwnedBy(HazelcastInstance instance, long lastKey, int increment) {
+        if (increment < 1) {
+            throw new IllegalArgumentException("increment must be at least 1");
+        }
+        Member localMember = instance.getCluster().getLocalMember();
+        PartitionService partitionService = instance.getPartitionService();
+        long key = lastKey;
+        for (int retries = 0; retries < 100; retries++) {
             Partition partition = partitionService.getPartition(key);
             if (localMember.equals(partition.getOwner())) {
                 return key;
             }
-            key++;
+            key += increment;
         }
+        throw new RuntimeException("Could not find next key for " + lastKey + " with increment " + increment);
```

Comment:
could not find key owned by instance, in the range lastKey to MAX_RETRIES inclusive